octopoes/octopoes/api/bulk_router.py

```diff
@@ -0,0 +1,52 @@
+import uuid
+from datetime import datetime
+
+import structlog
+from fastapi import APIRouter, Depends
+
+from octopoes.api.router import extract_valid_time
+from octopoes.api.router import settings as extract_settings
+from octopoes.config.settings import QUEUE_NAME_OCTOPOES, Settings
+from octopoes.core.app import get_xtdb_client
+from octopoes.events.manager import EventManager
+from octopoes.models.ooi.reports import HydratedReport
+from octopoes.repositories.ooi_repository import XTDBOOIRepository
+from octopoes.tasks.app import app as celery_app
+from octopoes.xtdb.client import XTDBSession
+
+logger = structlog.get_logger(__name__)
+router = APIRouter()
+
+
+@router.post("/reports", tags=["Reports"])
+def list_reports(
+    reports_filters: list[tuple[str, uuid.UUID]],
+    settings_: Settings = Depends(extract_settings),
+    valid_time: datetime = Depends(extract_valid_time),
+) -> dict[uuid.UUID, HydratedReport]:
+    """
+    An efficient endpoint for getting reports across organizations
+
+    The reason for creating the event_manager do this in the loop is the '_try_connect()' call in the __init__ possibly
+    slowing this down, while this API was introduced to improve performance. Simply reusing it for all clients works
+    because the event manager is only used in callbacks triggered on a `commit()`, while these queries are read-only and
+    hence don't need a `commit()` as no events would be triggered. (A cleaner solution would perhaps be to extract an
+    interface and pass a new NullManager.)
+
+    The xtdb_http_client is also created outside the loop and the `_client` property changed inside the loop instead,
+    to reuse the httpx Session for all requests.
+    """
+
+    event_manager = EventManager("null", str(settings_.queue_uri), celery_app, QUEUE_NAME_OCTOPOES)
+    xtdb_http_client = get_xtdb_client(str(settings_.xtdb_uri), "")
+
+    reports = {}
+
+    for client, recipe_id in reports_filters:
+        xtdb_http_client._client = client
```

Comment:
Accessing a private attribute like this is not Pythonic. Perhaps this attribute can now be a public property instead?

IMO *client* here is a bit ambiguous. Maybe not for now, but I suggest eventually renaming either the attribute or variable name to `organization_code`

Reply:
Although I understand your point, I'm not sure if I fully agree. I think having `organization` as a property of the client makes sense, similar to e.g. credentials or a base URL. I do agree that a proper refactor requires more thought. To keep the scope small, let's go with your suggestion for now while keeping in mind that we'll need to refactor (and rename) this at some point